ddtrace/bootstrap/sitecustomize.py

```diff
@@ -2,30 +2,112 @@
 Bootstrapping code that is run when using the `ddtrace-run` Python entrypoint
 Add all monkey-patching that needs to run by default here
 """
-import logging
-import os
 import sys
-from typing import Any
-from typing import Dict
 
 
-# Perform gevent patching as early as possible in the application before
-# importing more of the library internals.
-if os.environ.get("DD_GEVENT_PATCH_ALL", "false").lower() in ("true", "1"):
-    import gevent.monkey
+MODULES_LOADED_AT_STARTUP = frozenset(sys.modules.keys())
+MODULES_THAT_TRIGGER_CLEANUP_WHEN_INSTALLED = ("gevent",)
 
-    gevent.monkey.patch_all()
 
+import os  # noqa
+
+
+MODULES_TO_NOT_CLEANUP = {"atexit", "asyncio", "attr", "concurrent", "ddtrace", "logging"}
+if sys.version_info <= (2, 7):
+    MODULES_TO_NOT_CLEANUP |= {"encodings", "codecs"}
+    import imp
+
+    _unloaded_modules = []
+else:
+    import importlib
+
+
+def is_installed(module_name):
+    # https://stackoverflow.com/a/51491863/735204
+    if sys.version_info >= (3, 4):
```

Comment:
we use `importlib.metadata` in `ddtrace.internal.packages`, could that approach be applied here as well?

my thoughts:

- using same patterns we already have are good
- using `importlib.metadata` to get all distributions is heavier than just checking for the existence for a single package
    - we have caching enabled in `ddtrace.internal.packages` for this reason, which wouldn't get reused here

Reply:
I'm not sure I totally understand. If calling `importlib.metadata` here is heavier than the current approach, which I think means more computationally intensive, why would that be preferable? Maybe the answer is that following an existing pattern is worth the extra computation. Am I on the right track @brettlangdon?

Reply:
I think I am asking that question too 😆 

I don't think I expanded on desired action items/response. I am asking that same question to see if it was considered and if my understanding seems correct/valid. I am not saying it _should_ be changed.

Reply:
Thanks for clarifying!